source/manuals/programming-languages/java.html.md.erb

```diff
@@ -32,14 +30,16 @@ The IntelliJ "Optimize imports" command helps to sorting them alphabetically and
 
 ## Code checking
 
-Use of [Sonar](https://www.sonarqube.org/) for static analysis of code is encouraged.
+The use of static analysis is encouraged. Static analysis tools for Java include [SonarQube](https://www.sonarqube.org/), [Codacy](https://www.codacy.com/), [FindBugs](http://findbugs.sourceforge.net/) and [CheckStyle](http://checkstyle.sourceforge.net/). However, be aware that such tools can detect an overwhelming number of problems if applied to an existing project, which tends to result in their checks being ignored.
 
-## Build tools / external dependencies
+Try to minimise compiler warnings. If you cannot remove a warning, use an appropriate annotation to suppress it, preferably with a comment explaining why.
```

Comment:
Would be good to have a code snippet showing a good example of this

Reply:
Something like this?

```java
public void frobulate() {
    LOGGER.info("Frobulating foos");

    // LibFoo returns raw list but every element is always a Foo
    @SuppressWarnings("unchecked")
    List<Foo> foos = (List<Foo>) fooService.getFoos();

    foos.forEach(foo -> foo.frobulate());
}
```

Reply:
Looks good to me although shouldn't that comment go in the argument there instead of "unchecked"?

Reply:
I’m not sure I follow. Could you give a code snippet to illustrate what you mean?

Reply:
as in:

```java
    @SuppressWarnings("LibFoo returns raw list but every element is always a Foo")
```

Reply:
Ah, you can’t do that. The compiler actually understands `"unchecked"` to mean something (for `@SuppressWarnings`, the Java Language Specification defines `"unchecked"` and `"deprecated"` but other tools can support other strings). There’s no way to put an explanatory comment inside the annotation; it has to go in a comment outside.

Reply:
ahhh ok. I think I misunderstood this as our codebase does use other strings in places. Can we include the snippet you posted above then?

Reply:
Pushed another commit with the `@SuppressWarnings` example